api/config.storageos.com/v1/validate.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2022 Ondat.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1
+
+import (
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func (c *MetricsExporterConfig) Default() *MetricsExporterConfig {
+	if c.LogLevel == "" {
+		c.LogLevel = "info"
+	}
+	if c.Timeout == 0 {
+		c.Timeout = 10
+	}
+	return c
+}
+
+func (c *MetricsExporterConfig) Validate() error {
```

Comment:
It's a good practice to always validate user input however I would suggest instead of handling it on this level we add a kubebuilder validation to the StorageOSCluster resource.
Since the recent update where the field:
```
spec:
  metrics:
    enabled: true
```
was changed from a map[string]string into actual types you should be able to add something like 
`// +kubebuilder:validation:...` to those types and that way the users get immediate feedback when running `k edit storageoscluster` and trying to use the wrong type/value like -1 timeout
PS: what I am suggesting is on the operator repo, in the api folder

Reply:
The validation is being moved to the operator.